### ISSUE TITLE
10$: error on destructured optional parameter

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#855]
The following is invalid TypeScript:
```typescript
function f({}?) {}
```
However, the following is **valid** TypeScript:
```typescript
interface I {
  f({}?);
}
```
Report a diagnostic for the former but not the latter.
